feat: Add support for Sox's compression(bitrate) argument

adinhodovic opened the pull request:
Adds support for bitrate as parameter to output arguments. It is useful converting files to MP3 format. Requires a new version of soundfile(0.11.0) that supports MP3 formats alongside upgraded libsndfile version to 1.1.0. This is for the `test_bitrate_valid` test.

adinhodovic committed:
feat: Add support for Sox's compression(bitrate) argument

Adds support for bitrate as parameter to output arguments. It is useful converting files to MP3 format. Requires a new version of soundfile(0.11.0) that supports MP3 formats alongside upgraded libsndfile version to 1.1.0. This is for the `test_bitrate_valid` test.

lostanlen commented:
Hello @adinhodovic, the new feature looks great but from what I can see you are changing one of the dependencies from `pysoundfile` to `soundfile`. Can you remind me what happened there? Is this because of dependency resolution?

adinhodovic commented:
hey @lostanlen, I believe the dependency was renamed in the PyPi registry and the newer versions were only found under `soundfile`.

```
Breaking Changes
The soundfile module has evolved rapidly in the past. Most notably, we changed the import name from import pysoundfile to import soundfile in 0.7. In 0.6, we cleaned up many small inconsistencies, particularly in the the ordering and naming of function arguments and the removal of the indexing interface.
```

danihodovic commented:
It would be nice to have this merged @lostanlen

lostanlen commented on sox/transform.py:
```diff
@@ -343,6 +344,9 @@ def _validate_output_format(self, output_format):
             raise ValueError('channels must be a positive number')
 
+        if not isinstance(bitrate, float) and bitrate is not None:
+            raise ValueError('bitrate must be an float or None')
```
grammar

`must be a positive float` ?

adinhodovic replied:
:+1:

adinhodovic committed:
fix: Grammar issues

lostanlen commented:
tests are failing on MP3 ?

adinhodovic committed:
ci: Add libsox

adinhodovic commented:
Can you re run the workflow, think the OS mp3 dependency was missing.

grantgeorge commented:
👍 looking forward to this functionality being added
